assets/js/googlesitekit/modules/create-info-store.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * core/modules data store: info.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME as CORE_SITE } from '../datastore/site/constants';
+import { STORE_NAME as CORE_USER } from '../datastore/user/constants';
+
+const { createRegistrySelector } = Data;
+
+/**
+ * Creates a store object that has selectors for managing site info.
+ *
+ * @since 1.6.0
+ * @private
+ *
+ * @param {Object}  options               Options to consider for the store.
+ * @param {number}  options.storeName     Store name to use.
+ * @param {string}  options.adminPage     Store admin page. Default is 'googlesitekit-dashboard'.
+ * @param {boolean} options.requiresSetup Store flag, for requires setup. Default is 'true'
+ * @return {Object} The info store object.
+ */
+export const createInfoStore = ( {
+	storeName = undefined,
+	adminPage = undefined,
+	requiresSetup = undefined,
+} = {} ) => {
+	const STORE_NAME = storeName;
+
+	const INITIAL_STATE = {};
+	const actions = {};
+	const controls = {};
+	const reducer = ( state ) => {
+		return { ...state };
+	};
+	const resolvers = {};
+	const selectors = {
+		/**
+		 * Returns admin screen url.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param {(Object|undefined)} queryArgs Query arguments to add to admin URL.
+		 * @return {(string|undefined)} The admin screen url.
+		 */
+		getAdminScreenURL: createRegistrySelector( ( select ) => ( queryArgs ) => {
+			return select( CORE_SITE ).getAdminURL( adminPage, queryArgs );
+		} ),
+
+		/**
+		 * Returns admin reauth url.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @return {(string|undefined)} The admin reauth url.
+		 */
+		getAdminReauthURL: createRegistrySelector( ( select ) => () => {
+			const { needsReauthentication } = select( CORE_USER ).getAuthentication() || {};
+
+			const noSetupQueryArgs = ! requiresSetup ? {
+				notification: 'authentication_success',
+				reAuth: undefined,
+			} : {};
+
+			if ( ! needsReauthentication ) {
+				return select( STORE_NAME ).getAdminScreenURL( noSetupQueryArgs );
+			}
+
+			return select( CORE_SITE ).getAdminScreenURL();
```

Comment:
The logic in this method is not quite right. It needs to match _exactly_ what the current `getReAuthURL()` utility function does, except having the specifics being applied based on `adminPage` and `requiresSetup`. Some issues here are:
* Missing query parameters `slug` and `reAuth` (the `status` value from the original `getReAuthURL` function should always be `true` here, since we only use this URL for module activations as of now, never deactivations).
* Missing usage of the `connectURL`: The URL that's being built here should be like the `redirect` variable in the original `getReAuthURL` function; if `! needsReauthentication`, it should be returned as is, otherwise it should be added as `redirect` query parameter to the `connectURL` (with similar encoding etc. like in the original function), alongside `status: true`.
* This function should never return the general `select( CORE_SITE ).getAdminScreenURL()`.

Reply:
@felixarntz could I clarify this change with you. 

I've added `reAuth` to the parameter list for `getAdminReauthURL`, status is obtained as it is passed into the store. These 2 arguments are then passed to `getAdminScreenURL`.

My question surrounds the use of `connectURL or `adminURL` as the original function `getReAuthURL` required these as it was building the entire URL string but the new `getAdminURL` doesn't require them as it builds the base url and and adds the args to it. So my question is do we actually need `connectURL or `adminURL`?

Reply:
@ShahAaron We need `getAdminURL` to build the URL to the actual module page (or dashboard). But then, if `needsReauthentication` is `true`, we also need the connect URL (see `getReAuthURL` function's last few lines). There's the `getConnectURL` selector in `core/user` to retrieve that, and we'd need to pass the original URL built as `redirectURL` argument there.

Reply:
I have refactored this to include `connectURL` where the `needsReauthentication` is true